### ISSUE TITLE
Problem with foreign key configuration, 

### ISSUE THREAD

**Issue by msmolka** [aspnet/EntityFrameworkCore#10918]
All is working however I'm seeing a lot of strange information in debug log:

```
Microsoft.EntityFrameworkCore.Model: Information: The best match for foreign key properties {'VenueId' : Nullable<Guid>} are incompatible with the principal key {'TempId' : int}.
Microsoft.EntityFrameworkCore.Model: Information: The best match for foreign key properties {'VenueId' : Nullable<Guid>} are incompatible with the principal key {'TempId' : int}.
Microsoft.EntityFrameworkCore.Model: Information: The best match for foreign key properties {'VenueId' : Nullable<Guid>} are incompatible with the principal key {'TempId' : int}.
Microsoft.EntityFrameworkCore.Model: Information: The best match for foreign key properties {'VenueId' : Nullable<Guid>} are incompatible with the principal key {'TempId' : int}.
Microsoft.EntityFrameworkCore.Model: Information: The best match for foreign key properties {'VenueId' : Nullable<Guid>} are incompatible with the principal key {'TempId' : int}.
Microsoft.EntityFrameworkCore.Model: Information: The best match for foreign key properties {'VenueId' : Nullable<Guid>} are incompatible with the principal key {'TempId' : int}.
Microsoft.EntityFrameworkCore.Model: Information: The best match for foreign key properties {'VenueId' : Nullable<Guid>} are incompatible with the principal key {'TempId' : int}.
Microsoft.EntityFrameworkCore.Model: Information: The best match for foreign key properties {'VenueId' : Nullable<Guid>} are incompatible with the principal key {'TempId' : int}.
Microsoft.EntityFrameworkCore.Model: Information: The best match for foreign key properties {'VenueId' : Nullable<Guid>} are incompatible with the principal key {'TempId' : int}.
```

And it happens for all my models, so I have hundreds of similar lines.

### Steps to reproduce
I have only 1 model with `VenueId` foreign key:

```c#
public class Site
 {
        public Guid Id { get; set; }
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
        public string Name { get; set; }
        public virtual Venue Venue { get; set; }
        public Guid? VenueId { get; set; }
}

public class Venue
{
        public Guid Id { get; set; }
        public string Name { get; set; }
        public virtual IList<Site> Sites { get; set; }
}
```

and model builders:
site:
```c#
            builder.ToTable("Site");
            builder.HasKey(u => u.Id);
            builder.Property(r => r.Id).ValueGeneratedOnAdd();

            builder.Property(u => u.Name).HasMaxLength(100).IsRequired();
            builder.Property(u => u.Latitude).HasColumnType("numeric(9, 6)");
            builder.Property(u => u.Longitude).HasColumnType("numeric(9, 6)");

            builder.HasOne(a => a.Venue).WithMany(s => s.Sites).HasForeignKey(a => a.VenueId);
```
venue:
```c#
            builder.ToTable("Venue");
            builder.HasKey(u => u.Id);
            builder.Property(r => r.Id).ValueGeneratedOnAdd();

            builder.Property(u => u.Name).HasMaxLength(100).IsRequired();

            builder.HasMany(v => v.Sites).WithOne(s => s.Venue);
```

### Further technical details
EF Core version: 2.1.0-preview2-30045
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10x64
IDE: (e.g. Visual Studio 2017 15.5.6)
.NET Framework 4.6.2

**Comment by ajcvickers** [comment]
Try to log less and/or not messages with things like "TempId" that the user doesn't know about.

**Comment by msmolka** [comment]
I'm not worried too much about logs (it is only debug system), But I would like to have proper configuration to get rid of all warnings. 

For me such message means I did something wrong, so I'd like to know how to fix it.

**Comment by AndriySvyryd** [comment]
@msmolka The messages in `Information` and `Debug` level are not about something being wrong, but about showing what EF is doing under the cover. However in this case the messages aren't useful since they refer to an transient model state that occurs before any of your configuration is applied so there isn't much you can do to stop them from appearing.

**Comment by msmolka** [comment]
@AndriySvyryd but if it will happen always, what information it gives?
Shouldn't be such messages lowered to `Debug` level.
Information is useful for external developer  to trace queries and models.
If there is nothing to do, it should be `Debug` in my opinion.

**Comment by alexk8** [comment]
I've found this issue, because I've run into a bit similar problem.
I've turned on verbose mode and I've got lots of messages:
 _[DBG] The foreign key properties haven't been configured by convention because the best match {'Id' : int} are incompatible with the current principal key {'Id' : string}. ...'_. 
The problem is that I do not understand what entities are involved there, I can only guess that they are probably from Identity template. That's not really a problem, but I think there more info may be added (entity-types).
